recipes/catch2/2.x.x/conanfile.py

```diff
@@ -30,109 +32,104 @@ class Catch2Conan(ConanFile):
         "default_reporter": None,
     }
 
-    exports_sources = "CMakeLists.txt"
-    generators = "cmake"
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
-
     @property
     def _default_reporter_str(self):
-        return '"{}"'.format(str(self.options.default_reporter).strip('"'))
+        return str(self.options.default_reporter).strip('"')
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if not self.options.with_main:
-            del self.options.fPIC
-            del self.options.with_benchmark
+            self.options.rm_safe("fPIC")
+            self.options.rm_safe("with_benchmark")
 
     def package_id(self):
         if not self.options.with_main:
-            self.info.header_only()
+            self.info.clear()
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def validate(self):
-        if tools.Version(self.version) < "2.13.1" and self.settings.arch == "armv8":
-            raise ConanInvalidConfiguration("ARMv8 is supported by 2.13.1+ only! give up!")
-        if self.options.with_main and tools.Version(self.version) < "2.13.4":
-            raise ConanInvalidConfiguration("Option with_main not supported with versions < 2.13.4")
+        if Version(self.version) < "2.13.1" and self.settings.arch == "armv8":
+            raise ConanInvalidConfiguration("ARMv8 is not supported by versions < 2.13.1+")
+        if self.options.with_main and Version(self.version) < "2.13.4":
+            raise ConanInvalidConfiguration("Option with_main not supported by versions < 2.13.4")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
-
-    @functools.lru_cache(1)
-    def _configure_cmake(self):
-        cmake = CMake(self)
-        cmake.definitions["BUILD_TESTING"] = "OFF"
-        cmake.definitions["CATCH_INSTALL_DOCS"] = "OFF"
-        cmake.definitions["CATCH_INSTALL_HELPERS"] = "ON"
-        cmake.definitions["CATCH_BUILD_STATIC_LIBRARY"] = self.options.with_main
-        cmake.definitions["enable_benchmark"] = self.options.get_safe("with_benchmark", False)
```

Comment:
what did happen to `enable_benchmark` ?

Reply:
There is no such symbol anywhere in catch2's 2.x.x source code!
Or in 3.x.x
I did conan create on a bunch of versions and grepped the whole cache - nothing found!

Reply:
note that the option with_benchmark IS used for CATCH_CONFIG_ENABLE_BENCHMARKING